PSReadLine/Replace.vi.cs

```diff
@@ -151,16 +151,12 @@ private static void ViReplaceWord(ConsoleKeyInfo? key, object arg)
                 }
                 _singleton._lastWordDelimiter = char.MinValue;
             }
-            if (_singleton._current == _singleton._buffer.Length - 1
-                && !_singleton.IsDelimiter(_singleton._lastWordDelimiter, _singleton.Options.WordDelimiters)
-                && _singleton._shouldAppend)
-            {
-                ViInsertWithAppend(key, arg);
-            }
-            else
-            {
-                ViInsertMode(key, arg);
-            }
+
+            ViInsertModeAndForwardChar(
+                () => (_singleton._current == _singleton._buffer.Length - 1
+                    && !_singleton.IsDelimiter(_singleton._lastWordDelimiter, _singleton.Options.WordDelimiters)
+                    && _singleton._shouldAppend),
```

Comment:
Why is the lambda needed? Why not calculate the bool result here and pass that in?

Reply:
Good catch, no real reason. I will refactor.